### PR TITLE
GAMBoost

### LINKED ISSUE #38: Connect learner gamboost from package mboost
[CRAN](https://cran.r-project.org/web/packages/mboost/index.html)

### PR DESCRIPTION
**Pull request by be-marc**
Closes mlr-org/mlr3learners#38

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,6 +20,7 @@
 *.utf8.md
 *.knit.md
 inst/doc
-
 .DS_Store
 .Rproj.user
+.Rprofile
+renv
```
Comment (mllg) [inline .gitignore:26]: IDK what renv is, but .Rprofile could go into the .Rbuildignore of the template repository.

File: R/LearnerClassifGAMBoost.R [hunk 2]
```diff
@@ -0,0 +1,99 @@
+#' @title Boosted Generalized Additive Classification Learner
+#'
+#' @aliases mlr_learners_classif.gamboost
+#' @format [R6::R6Class] inheriting from [LearnerClassif].
+#'
+#' @description
+#' A [LearnerClassif] for a classification gamboost implemented in [mboost::gamboost()] in package \CRANpkg{mboost}.
```
Comment (mllg) [inline R/LearnerClassifGAMBoost.R]: Don't you need to link with `[mlr3::LearnerClassif]`?

File: R/LearnerClassifGAMBoost.R [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      withr::with_package("mboost", { # baselearner argument requires attached mboost package
+        invoke(mboost::gamboost, formula = f, data = data, control = ctrl, .args = pars_gamboost)
+      })
+    },
+
+    predict_internal = function(task) {
+      family = self$param_set$values$family
+      newdata = task$data(cols = task$feature_names)
+
+      if (self$predict_type == "prob" & (family == "AdaExp" | family == "AUC")) {
```
Comment (mllg) [inline R/LearnerClassifGAMBoost.R]: `&&` and `||`, these are scalar values

File: R/LearnerRegrGAMBoost.R [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+#' @export
+LearnerRegrGAMBoost = R6Class("LearnerRegrGAMBoost", inherit = LearnerRegr,
+  public = list(
+    initialize = function() {
+      ps = ParamSet$new(
+        params = list(
+          ParamFct$new(id = "baselearner", default = "bbs", levels = c("bbs", "bols", "btree"), tags = c("train")),
+          ParamInt$new(id = "dfbase", default = 4L, tags = c("train")),
+          ParamDbl$new(id = "offset", default = NULL, special_vals = list(NULL), tags = c("train")),
+          ParamFct$new(id = "family", default = c("Gaussian"),
```
Comment (mllg) [inline R/LearnerRegrGAMBoost.R:24]: Try to avoid concatenating scalar values.

**Review by mllg — CHANGES_REQUESTED** [review]

File: R/LearnerRegrGAMBoost.R (hunk 4 above)
Comment (be-marc) [reply R/LearnerRegrGAMBoost.R:24]: Forgot to change this after removing some tags

File: R/LearnerClassifGAMBoost.R (hunk 2 above)
Comment (be-marc) [reply R/LearnerClassifGAMBoost.R]: Works without it but changed it for clarity

File: .gitignore (hunk 1 above)
Comment (be-marc) [reply .gitignore:26]: renv is the new packrat

**Review by mllg — APPROVED** [review]

**Pull request merged**